### DISCUSSION TITLE
[QUESTION]: rollupOptions manualChunks

### DISCUSSION THREAD

**Discussion by simplecommerce** [vikejs/vike#1421] [Discussion] [answered]
Hi,

I was having some issues on my build for a test project I have and noticed that in order to fix my build I had to use rollupOptions manualChunks for specific imports.

All seems to be fine, but I noticed that it doesn't seem to change the output of the files being created as mentioned in the documents and examples I have seen online.

I tried using this repo mentioned in your docs for MUI https://github.com/mikew/vike-example-mui-emotion-cache.
And I did a simple test like this: 

```
  build: {
    // target: 'esnext', //browsers can handle the latest ES features
    // minify: false,
    rollupOptions: {
      output: {
        manualChunks: id => {
          if (id.includes("@mui/icons-material/Menu")) {
            console.log(id)
            return "mui-menu";
          }
        }
      }
    }
  },
```

I do see the output in my console but I don't see a new file or chunk called "mui-menu".

The test that I saw online was on this example: https://stackblitz.com/edit/some-random-chunking-mechanism?file=vite.config.ts

If I run the npm run build, based on their config I do see the `index` and `nested` chunks.

I assume this is because vike handles it differently? 
I didn't see this mentioned anywhere in the docs so I figured i'd ask, thanks!

**Comment by brillout** [comment, marked as answer]
Vike doesn't interfere with chunking. That said, Vike does modify the filenames of chunks, which explains why you don't see them although there are actually here. (It's something we should document at some point.) If you want, you can debug and remove the relevant bits of `node_modules/vike/dist/esm/node/plugin/plugins/distFileNames.js` and see for yourself.

There is probably a feature request looming around, e.g. for enabling aggressive caching of vendor assets. In the meantime, you can override Vike `rollupOutput.chunkFileNames` in the user land.

**Reply by simplecommerce** [reply]
Thanks for your reply, I simply asked because in my tests I wasn't sure of the behavior.
I had an issue where I was getting an error with a function being called/used before it was declared within the same file because of the ordering. 
So I tested with the manualChunks to fix it, I was able to fix it but only by returning the id of the file for each of my custom files in my library.

this worked
```
  build: {
    // target: 'esnext', //browsers can handle the latest ES features
    // minify: false,
    rollupOptions: {
      output: {
        manualChunks: id => {
          if (id.includes("/lib")) {
            return id;
          }
        }
      }
    }
  },  
```

this didn't work for some reason
```
  build: {
    // target: 'esnext', //browsers can handle the latest ES features
    // minify: false,
    rollupOptions: {
      output: {
        manualChunks: id => {
          if (id.includes("/lib")) {
            return "lib";
          }
        }
      }
    }
  },
```

so I was suspecting it had something to do with the fact that I couldn't see my file names being created.

I will play around with it with what you mentioned, thanks again!

**Reply by brillout** [reply]
I guess it's a Rollup bug? In principle, I would assume that `manualChunks` cannot break the app (albeit significantly degrading performance).

I pushed a [little fix](https://github.com/vikejs/vike/commit/d07fbe827118677177ef1226afb5ba23fc86fd43) which is pre-released in `0.4.156-commit-d07fbe8`: the user can now suppress Vike's naming by, for example, setting `build.rollupOptions.output.chunkFileNames` to `null`.

Btw, any news on sponsoring? (IIRC you mentioned it could be an option.)

**Reply by simplecommerce** [reply]
Ok I will test it out, thank you!

Yes, I haven't forgotten, we have been very busy on our end prior the holidays and haven't had time to even jump back on the project, which is what I was doing this week but kept getting interrupted due to various issues like the ones you just helped me with.

As soon as I can complete my tests and we have the go ahead, I will get back to you about sponsoring, thanks!

**Reply by brillout** [reply]
Thanks, we'd hugely appreciate it 💚
